packages/react-chat/src/hooks/useRuntime.ts

```diff
@@ -45,6 +46,32 @@ const DEFAULT_RUNTIME_STATE: Required<SessionOptions> = {
 export const useRuntime = ({ url = RUNTIME_URL, versionID, verify, user, ...config }: UseRuntimeProps) => {
   const [indicator, setIndicator] = useState(false);
   const [session, setSession, sessionRef] = useStateRef<Required<SessionOptions>>({ ...DEFAULT_RUNTIME_STATE, ...config.session });
+  const [lastInteractionAt, setLastInteractionAt] = useState<number | null>(Date.now());
+  const [noReplyTimeout, setNoReplyTimeout] = useState<number | null>(null);
+
+  useEffect(() => {
+    let noReplyTimer: NodeJS.Timeout | undefined;
+
+    const checkNoReply = () => {
+      const ready = isStatus(SessionStatus.ACTIVE);
+
+      if (ready && noReplyTimeout && lastInteractionAt) {
+        const timeSinceLastInteraction = Date.now() - lastInteractionAt;
+        if (timeSinceLastInteraction > noReplyTimeout) {
+          // Trigger no reply action
+          interact({ type: ActionType.NO_REPLY, payload: null });
```

Comment:
should we reset `noReplyTimeout` here to don't trigger no-reply multiple times?

Reply:
it seems like we don't need to. i added a log statement and confirmed that it's only running this line once each time.
i believe it's not necessary because calling `interact()` here will have it call `setLastInteractionAt()`, which has the same effect as resetting the no reply timeout.